js/src/evaluate.js

```diff
@@ -3,49 +3,74 @@
  * @author Nicholas C. Zakas
  */
 
+//-----------------------------------------------------------------------------
+// Typedefs
+//-----------------------------------------------------------------------------
+
+/** @typedef {import("./momoa").MomoaNode} MomoaNode */
+/** @typedef {import("./momoa").MomoaNodeParts} MomoaNodeParts */
+/** @typedef {import("./momoa").MomoaDocumentNode} MomoaDocumentNode */
+/** @typedef {import("./momoa").MomoaStringNode} MomoaStringNode */
+/** @typedef {import("./momoa").MomoaNumberNode} MomoaNumberNode */
+/** @typedef {import("./momoa").MomoaBooleanNode} MomoaBooleanNode */
+/** @typedef {import("./momoa").MomoaMemberNode} MomoaMemberNode */
+/** @typedef {import("./momoa").MomoaObjectNode} MomoaObjectNode */
+/** @typedef {import("./momoa").MomoaElementNode} MomoaElementNode */
+/** @typedef {import("./momoa").MomoaArrayNode} MomoaArrayNode */
+
 //-----------------------------------------------------------------------------
 // Exports
 //-----------------------------------------------------------------------------
 
 /**
  * Evaluates a Momoa AST node into a JavaScript value.
- * @param {Node} node The node to interpet.
+ * @param {MomoaNode} node The node to interpet.
```

Comment:
if you make MomoaAnyNode a discriminated union, you might be able to use it here, which would let you get rid of the casts inside the switch, because the type would be narrowed to the correct subtype in each branch.

Reply:
Interesting...I'll play around with that and see if I can get it working.